docker/templates/new-gatsby-app/{{cookiecutter.app_name}}/src/components/header.js

```diff
@@ -0,0 +1,42 @@
+import { Link } from "gatsby"
+import PropTypes from "prop-types"
+import React from "react"
+
+const Header = ({ siteTitle }) => (
+  <header
+    style={{
```

Comment:
Style question: We tend to avoid inline styles when writing HTML. What's our stance on inline styling for components?

Reply:
Great question, and one that I'd like to have a stronger stance on! (I suspect going through the course at https://mastergatsby.com/ would supply it)

[This section of the Gatsby tutorial](https://www.gatsbyjs.com/tutorial/part-two/) gives a good account of the different CSS approaches: global styles, component-scoped styles, and CSS-in-JS (aka inline styles).

I've used all 3 at different times, and the approach that I've liked most has been to have a global style sheet (which this repo doesn't include, but maybe should—[here](https://gitlab.com/court-transparency-project/charges-app/-/blob/master/src/components/styles.css) it is on the courts app) and to handle the component-level styling with CSS-in-JSS. The Gatsby docs also recommend using CSS modules, but I've found it cumbersome to have to create a second CSS file to update alongside every component.

I've also used the `styled-components` package for CSS-in-JS, and I'd originally included it in this template, but I haven't always actually found it necessary. In courts, David started his components with it and I've ended up writing it out.

This is a good overview of the benefits of CSS-in-JSS: https://medium.com/seek-blog/a-unified-styling-language-d0c208de2660

I'm inclined to add a global style sheet to this repo for now, and plan to come back to this question later and flesh out official guidance on DataMade's React CSS practices if I go through that Mastering Gatsby class. I agree that avoiding inline CSS styles in Django projects is still a best practice.

Reply:
That's a good plan! I agree it'd be nice to have an example of global styles in the template.